### PR TITLE
Add event_class_map for event class aliases

### LINKED ISSUE #177: Allow aliases for stored event classes instead of persisting FQCN to DB
Similar to how you can set an alias with Relation::morphMap() when using polymorphic eloquent relationships, it would be nice to be able to do something similar for how 'event_class' is saved in the stored events table. This way you could change namespaces and folder structure without affecting previously saved events.

### PR DESCRIPTION
**Pull request by riasvdv**
Fixes #177

### CODE REVIEW

File: src/Models/StoredEvent.php [hunk 1]
```diff
@@ -27,7 +28,7 @@ public static function createForEvent(ShouldBeStored $event, string $uuid = null
     {
         $storedEvent = new static();
         $storedEvent->aggregate_uuid = $uuid;
-        $storedEvent->event_class = get_class($event);
+        $storedEvent->event_class = self::getEventClass(get_class($event));
```
Comment (freekmurze) [inline src/Models/StoredEvent.php]: Let's use `static` here.

File: src/Models/StoredEvent.php [hunk 2]
```diff
@@ -37,6 +38,11 @@ public static function createForEvent(ShouldBeStored $event, string $uuid = null
         return $storedEvent;
     }
 
+    public function getEventClassAttribute(string $value): string
+    {
+        return self::getActualClassForEvent($value);
```
Comment (freekmurze) [inline src/Models/StoredEvent.php]: Use `static`

File: src/Models/StoredEvent.php [hunk 3]
```diff
@@ -100,4 +106,20 @@ public static function store(ShouldBeStored $event, string $uuid = null): void
     {
         static::storeMany([$event], $uuid);
     }
+
+    private static function getEventClass(string $class): string
```
Comment (freekmurze) [inline src/Models/StoredEvent.php]: Let's use `protected`

File: src/Models/StoredEvent.php [hunk 4]
```diff
@@ -100,4 +106,20 @@ public static function store(ShouldBeStored $event, string $uuid = null): void
...
+        $map = config('event-projector.event_class_map', []);
+
+        if (! empty($map) && in_array($class, $map)) {
+            return array_search($class, $map, true);
+        }
+
+        return $class;
+    }
+
+    private static function getActualClassForEvent(string $class): string
```
Comment (freekmurze) [inline src/Models/StoredEvent.php]: Let's use `protected`

File: src/Models/StoredEvent.php [hunk 5]
```diff
@@ -38,9 +38,9 @@ public static function createForEvent(ShouldBeStored $event, string $uuid = null
         return $storedEvent;
     }
 
-    public function getEventClassAttribute(string $value): string
+    public static function getEventClassAttribute(string $value): string
```
Comment (freekmurze) [inline src/Models/StoredEvent.php]: I'm thinking this one doesn't need to be static.

Comment (riasvdv) [reply src/Models/StoredEvent.php]: Done!

**Comment by freekmurze** [comment]
Perfect, thanks!

**Pull request merged**